lib/graphql/schema/default_type_error.rb

```diff
@@ -0,0 +1,19 @@
+module GraphQL
+  class Schema
+    module DefaultTypeError
+      def self.call(value, field, parent_type, query_ctx)
+        if field.type.kind.non_null?
+          if value.nil?
+            query_ctx.errors << GraphQL::InvalidNullError.new(parent_type.name, field.name, value)
+          else
+            # it was caused by GraphQL::ExecutionError
```

Comment:
The type_error hook isn't called for a GraphQL::ExecutionError, nor should it.  So I think this is just an unused code path, since this hook is only being used for invalid null and unresolved type errors.